docs/docs/introduction/container-images.md

```diff
@@ -0,0 +1,204 @@
+
+ - are encapsulations of an environment
+ - have their own process namespace
+ - the main container on the host is still associated with one process, and when you start that process the set within the container might kick off, and when you kill it the container's processes are also killed.
+
+
+### Container Namespaces
+
+Akin to the namespaces on your host, a container also has its own namespace? 
+Here it is helpful to think of a container like a little packaged environment, one that you can actually shell into like you would a server. 
+When I shell inside the container I find that it has its own little process namespace.
+
+### Container Cgroups
+
+A cgroup is a "control group" and it's exactly that - a feature of the kernel that lets you allocate permissions for resources. 
+So, for example, I might allocate a particular amount of memory, or CPU time, or network bandwidth to a set of tasks, these processes (point to processes) that are running on my computer. 
+What else can we do? 
+We can monitor them, reconfigure them, and even deny them access to resources.
+So these containers also have their own set of cgroups.
+
+## What about a container image?
+
+The words "container" and "image" are sometimes used synonymously, but are subtly two different things. 
+You can think of a container image as a template, and a container is one or more running instances of that template. 
+In more concrete terms, the image is some binary sitting on your filesystem that is ready to instantiate to create an isolated, encapsulated environment.
+
+## Example Interactions
+
+Let's give a concrete example of how you might use a container. 
+Let's say you are a scientist, and a life changing paper just came out that describes how to estimate an optimal cheese eating time based on measured properties of the cheese. 
+You've been collecting cheeses for decades, you can easily collect the metrics needed, and you just need the software. 
+
+### 1. You find container images in a registry
+{: .no-toc }
+
+Thankfully, the publishes also released a container in a registry (such as [Docker Hub](https://hub.docker.com/)) and you can:
+
+ 1. Install the Docker daemon to interact with Docker containers
+ 2. Pull the container to your local machine
+ 3. Run the container with your input parameters
+
+That would it, and truly that easy!
+
+### 2. You want to change the cheese algorithm
+{: .no-toc }
+
+Uhoh, the scientist calculated the relationship between the moisture content of cheese completely wrong, and you need to modify his code and then write your own paper to refute his claims. 
+Dry cheese like cheddar deserves just as high of a rating! 
+In this case, you would:
+
+ 1. Install the Docker daemon to interact with Docker containers
+ 2. Find the container recipe (build specification), a Dockerfile, and modify his code
+ 3. Build and run your new container
```

Comment:
build your new container, test/run it on your computer until you get the correct result...